samples/nrf9160/modem_shell/src/link/link.c

```diff
@@ -524,21 +560,22 @@ void link_rsrp_subscribe(bool subscribe)
 }
 
 void link_ncellmeas_start(bool start, enum link_ncellmeas_modes mode,
-			  enum lte_lc_neighbor_search_type search_type,
+			  struct lte_lc_ncellmeas_params ncellmeas_params,
 			  int periodic_interval)
 {
 	int ret;
 
-	ncellmeas_mode = mode;
-	ncellmeas_periodic_interval = periodic_interval;
-	ncellmeas_search_type = search_type;
+	ncellmeas_param_data.mode = mode;
+	ncellmeas_param_data.periodic_interval = periodic_interval;
+	ncellmeas_param_data.params = ncellmeas_params;
```

Comment:
Is there a reason why we are not setting `ncellmeas_work_data` here? It could be that scheduling ncellmeas_work_data cause the use of `ncellmeas_param_data` here but as I don't know I'll add this comment.

Reply:
As discussed offline, I'll keep these as separate.